### DISCUSSION TITLE
Invalidate a session from database

### DISCUSSION THREAD

**Discussion by Gerald-Gmainer** [supabase/supabase#13941] [Questions] [answered]
Hi, I am using admin roles for users and using custom claims as described here
https://dev.to/supabase/supabase-custom-claims-34l2
My RLS is using those custom claims, which works almost perfectly.

But when the roles are changed, a manual logout is required to reload the roles (or waiting only session is refreshed). Which is also described in that tutorial.

I was hoping to do any changes in `auth.sessions` or  `auth.refresh_tokens`

I tried to 
```
DELETE from auth.sessions where user_id='user_id';
DELETE from auth.refresh_tokens where user_id='user_id';

UPDATE auth.sessions SET revoked = true WHERE user_id = 'user_id';
UPDATE auth.sessions SET not_after = NOW() WHERE user_id = 'user_id';
UPDATE auth.refresh_tokens SET revoked = true WHERE user_id = user_id';
```

But those queries could not force to refresh the session; let just let the user logout. I am also hoping to achieve that without realtime. 

Is it possible from database to refresh the session or let the user logout?

**Comment by GaryAustin1** [comment, marked as answer]
The clients have no connection to the auth server except doing REST http requests.   There is no way for the database or auth to communicate to them.   Even using signOut(), the jwt tokens on other devices are still good until they expire.   For this reason their expiration should be shorter based on your security needs.

**Reply by ghost** [reply]
@GaryAustin1 on a slightly separate but related note, would calling supabase.auth.**_admin_**.signOut() invalidate existing access tokens, or would those also be good for their duration and until expiration? I saw several conflicting discussions about the server-side auth.admin.signOut() behaving differently from the client side auth.signOut() and wasn't sure if this was the case.

**Reply by GaryAustin1** [reply]
It will only invalidate them being refreshed on other devices.  There is no check on an access token jwt other than is it expired which is built in the jwt.  The access token is not checked against any tables or other data from gotrue when it is used.  And as I said before the server cannot send a message to the client.

**Reply by ghost** [reply]
makes perfect sense; thank you very much @GaryAustin1 !

**Reply by ArmanAryanpour** [reply]
Is there an easy way in supabase to implement a JWT blocklist for anomaly detection and suspicious activities?

**Reply by GaryAustin1** [reply]
@ArmanAryanpour 
You would need a table with UUID's (in the JWT) to block.  Then in EVERY RLS policy have a check against that UUID.  I suspect this would be very painful on performance.
You do not have access to the JWT itself when the database is called, and there are no settings in Supabase for doing that in Kong(?) or some other spot.
